File: data/data.code.cpp.server/AckTestJsOk.cpp
#ifndef _ACK_TEST_JS_OK_
#define _ACK_TEST_JS_OK_

#include <list>

#include <pb_type.hpp>
#include <Packet.hpp>

#include <Msg.hpp>



class AckTestJsOk
{
private:
	U64 m_u64;
	I64 m_i64;


public:
	Packet Encode()
	{
		Packet packet;
		packet.WriteU64(m_u64);
		packet.WriteI64(m_i64);
		packet.Encode(Msg::P_ACK_TEST_JS_OK);
		return packet;
	}


	void SetU64(U64 u64)
	{
		m_u64 = u64;
	}

	void SetI64(I64 i64)
	{
		m_i64 = i64;
	}

};

#endif

File: data/data.code.lua.server/AckTestJsOk.lua
module("AckTestJsOk", package.seeall)


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = AckTestJsOk})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.u64 = 0
	self.i64 = 0
end


function encode(self)
	local pack = packet.new()
	pack:writeU64(self.u64)
	pack:writeI64(self.i64)
	return pack:encode(Msg.P_ACK_TEST_JS_OK)
end
